ext/standard/array.c

```diff
@@ -6358,7 +6358,7 @@ PHP_FUNCTION(array_key_exists)
 			RETVAL_BOOL(zend_hash_index_exists(ht, Z_RES_HANDLE_P(key)));
 			break;
 		default:
-			zend_type_error("Illegal offset type");
+			zend_argument_type_error(1, "to be a valid array offset type");
```

Comment:
Maybe change this to ``int|string`` ?

Reply:
`int|string` is not enough, because all the types except for `array` and `object` are supported

Reply:
Well I'm basing my self from the error message you wrote which expects an array offset apparently? Which can only be an integer or a string. But I didn't look at the surrounding code itself tbh.